core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java

```diff
@@ -448,9 +448,18 @@ private void recoverClusterStatus(int newViewId, final boolean isMergeView, fina
                         new CacheTopologyFilterReuser());
             break;
          } catch (ExecutionException e) {
-            if (i == 0 || !(e.getCause() instanceof TimeoutException))
-               throw e;
-            log.debug("Timed out waiting for cluster status responses, trying again");
+            if (i != 0) {
+               if (e.getCause() instanceof TimeoutException) {
+                  log.debug("Timed out waiting for cluster status responses, trying again");
+               } else if (e.getCause() instanceof SuspectException) {
+                  if (transport.getMembers().containsAll(clusterMembers)) {
+                     log.debug("Received a CacheNotFoundResponse from one of the members, trying again");
+                     Thread.sleep(getGlobalTimeout() / CLUSTER_RECOVERY_ATTEMPTS / 2);
```

Comment:
Hi @danberindei, do you recall why you sleep here? With default settings, it's 12 seconds, and therefore, during partition, when the node installs view with unreachable members and later a view with solely itself (as it became the only node in its partition), there's first view installation + 12s delay before it becomes degraded.